src/engine/irc.cpp

```diff
@@ -330,49 +330,61 @@ ICOMMAND(0, ircaddrelay, "ssisss", (const char *n, const char *s, int *p, const
 ICOMMAND(0, ircserv, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current server is: %s", n->serv); return; }
+    if(s == 0 || *s != '\0') { ircprintf(n, 4, NULL, "current server is: %s", n->serv); return; }
```

Comment:
-Wzero-as-null-pointer-constant will trigger here. Use either the inexpressive old way, or compare with nullptr. The same happens in other places.